fastcrypto/src/bls12377.rs

```diff
@@ -760,3 +760,92 @@ impl Drop for BLS12377KeyPair {
         self.zeroize();
     }
 }
+
+#[cfg(feature = "experimental")]
+pub mod mskr {
+    //
+    // Implement MSKR for BLS12377
+    //
+    use crate::bls12377::{
+        BLS12377KeyPair, BLS12377PrivateKey, BLS12377PublicKey, BLS12377Signature,
+        CELO_BLS_PUBLIC_KEY_LENGTH,
+    };
+    use crate::hash::HashFunction;
+    use crate::hash::Sha256;
+    use crate::traits::mskr::{HashToScalar, Randomize};
+    use ark_bls12_377::{Fr, G1Projective, G2Projective};
+    use ark_ec::group::Group;
+    use ark_ff::BigInteger256;
+    use celo_bls::{PrivateKey, PublicKey};
+    use once_cell::sync::OnceCell;
+    use std::ops::Mul;
+
+    pub struct BLS12377Hash {}
+
+    impl HashToScalar<Fr> for BLS12377Hash {
+        fn hash_to_scalar(bytes: &[u8]) -> Fr {
```

Comment:
we could use the same method that we generate private keys in BLS where some 32 bytes entropy bytes are an input, in our case the whole digest. However @asonnino it's very common for rogue key protection to use half scalars, because we care about pre-image attacks; you try to cancel-out another pub key, so combinatorics are already very expensive. 

We do use half scalars in many apps (ie in the EdDSA half aggregation) for performance reasons to make that exponentiation half-expensive. In fact though, here we don't care, as this applies once at private key randomization (and with our mode where new-priv_key = old_priv_key \times this_scalar, we don't even have an ECC operation). 

@asonnino: You remember the discussion we had at overleaf, shall we update the private key or for each signature we multiply by this scalar? if we do the former half or full scalar doesn't really matter. In the second you have to pay that cost every time you sign.

at this stage and while experimental, I'll approve the mode as it is. But @jonas-lj let's have a TODO.